flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/gsa/GatherSumApplyIteration.java

```diff
@@ -367,6 +394,20 @@ public void join(Vertex<K, VV> vertex, Edge<K, EV> edge, Collector<Tuple2<K, Nei
 		}
 	}
 
+	@SuppressWarnings("serial")
+	@ForwardedFieldsSecond("f1->f0")
```

Comment:
I think this annotation is wrong. It's the first field that's forwarded (the edge source).
If that's the case, can you please investigate why your tests don't catch this?